chain/network/src/peer/peer_actor.rs

```diff
@@ -1057,8 +1063,23 @@ impl PeerActor {
         .entered();
 
         match peer_msg.clone() {
-            PeerMessage::Disconnect => {
+            PeerMessage::Disconnect(reason) => {
                 tracing::debug!(target: "network", "Disconnect signal. Me: {:?} Peer: {:?}", self.my_node_info.id, self.other_peer_id());
+
+                match reason {
+                    ClosingReason::PeerManagerRequest | ClosingReason::Ban(_) => {
```

Comment:
please document why specifically those reasons should trigger removal from recent_connections